lib/python/ensembl/brc4/runnable/compare_fasta.py

```diff
@@ -212,127 +174,129 @@ def compare_seqs(self, seq1, seq2):
+        seq_data = self.get_json(map_dna_path)
+        org_loc = self.organellar_assembly(report_seq, seq_data)
+        INSDC_assembly_level, core_assembly_level = self.assembly_level(
+            report_seq, seq_data
+        )
 
-        only1_id = [name for seq, name in only1.items()]
-        only2_id = [name for seq, name in only2.items()]
+        comp.append(
+            "Assembly level: %s vs %s" % (INSDC_assembly_level, core_assembly_level)
+        )
 
-        for i in only1_id:
-            i = str(i)
-            for insdc, core in All_id.items():
-                if i == insdc and i != core:
-                    comp.append("%s and %s match by id" %
-                                (i, core))
-                else:
-                    pass
+        names_length = {}
+        # sequences which have extra N at the end
+        if only1 and only2:
+            for seq1, name1 in only1.items():
+                len1 = len(seq1)
+                seq1_N = seq1.count("N")
+                for seq2, name2 in only2.items():
+                    len2 = len(seq2)
+                    seq2_N = seq2.count("N")
+                    sequence_2 = seq2[:len1]
+                    if sequence_2 == seq1:
+                        ignored_seq = seq2[len1:]
+                        N = ignored_seq.count("N")
+                        if len(ignored_seq) == N:
+                            comp.append(
+                                f"Please check extra Ns added in core in  %s and %s"
+                                % (name1, name2)
+                            )
+                        else:
+                            comp.append(
+                                f"ALERT INSERTIONS at the end or diff assembly level  %s and %s"
+                                % (name1, name2)
+                            )
+                    elif len1 == len2:
+                        if seq2_N > seq1_N:
+                            comp.append(
+                                f"Core has more Ns, check  %s and %s" % (name1, name2)
+                            )
+                        elif seq1_N > seq2_N:
+                            comp.append(
+                                f"INSDC has more Ns, check  %s and %s" % (name1, name2)
+                            )
+                        else:
+                            names_length[name1] = name2
+                    else:
+                        continue
 
-        for id in only2_id:
-            id = str(id)
-            if id in All_id.values():
-                comp.append("%s and %s match by id" % (i, All_id[i]))
-            else:
-                pass
+        if names_length:
+            length = len(names_length)
+            comp.append(f"%d sequences have the same length" % length)
+            for insdc, core in names_length.items():
+                comp.append(f"INSDC: %s and coredb : %s" % (insdc, core))
+
+        # Remove the duplicates
+        for org_name in list(org_loc.keys()):
```

Comment:
```suggestion
        for org_name in org_loc.keys():
```
`keys()` returns an iterator, which is more efficient to loop than transforming all the keys into a list and then iterating over it.

Reply:
This gives me a "RuntimeError: dictionary changed size during iteration"  as I am altering  the dictionary in the loop

Reply:
Ahh missed that, sorry.